File: ruby/WeaponType.rb
module Deepspace
  module WeaponType

    # @brief  Representa a los tipos de armas del juego.
    #         cada valor posible del tipoenumerado tendrá asociado un valor
    #         numérico concreto igual a la potencia de disparo de cada tipode arma.
    class Type
      # consultor power
      attr_reader :power # equivalente a devolver @power

      def initialize ( _power)
        @power= _power
      end # initialize

    end # class Type

    LASER= Type.new(2.0)
    MISSILE= Type.new(3.0)
    PLASMA= Type.new(4.0)
  end # module weapon
end

File: java/deepspace/src/deepspace/WeaponType.java
package deepspace;


public enum WeaponType { 
    LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f); // f para pasar de double a float

    private float power; 

    //constructor 
    WeaponType ( float power){
	this.power = power; 
    }

    // Consultor
    float getPower(){
	return power; 
    }
    
    public String toString(){
        if(power==2.0){
            return "LASER";
        }
        if(power == 3.0){
            return "MISSILE";
        }
        if(power == 4.0) {
            return "PLASMA";
        }
        return "Error";
    }

} //enum WeaponType
